src/commands/altimateScan.ts

```diff
@@ -0,0 +1,150 @@
+import { Diagnostic, DiagnosticSeverity, Range, Uri } from "vscode";
+import { AltimateRequest } from "../altimate";
+import { DBTProjectContainer } from "../manifest/dbtProjectContainer";
+import {
+  ManifestCacheChangedEvent,
+  ManifestCacheProjectAddedEvent,
+} from "../manifest/event/manifestCacheChangedEvent";
+import { TelemetryService } from "../telemetry";
+import { provideSingleton } from "../utils";
+import { findModelProblems } from "./command_utils";
+import { DBTProject } from "../manifest/dbtProject";
+
+export interface AltimateCatalog {
+  [projectName: string]: { [key: string]: any[] };
+}
+
+@provideSingleton(AltimateScan)
+export class AltimateScan {
+  private eventMap: Map<string, ManifestCacheProjectAddedEvent> = new Map();
+  constructor(
+    private dbtProjectContainer: DBTProjectContainer,
+    private telemetry: TelemetryService,
+    private altimate: AltimateRequest,
+  ) {
+    dbtProjectContainer.onManifestChanged((event) =>
+      this.onManifestCacheChanged(event),
+    );
+  }
+
+  private async onManifestCacheChanged(event: ManifestCacheChangedEvent) {
+    event.added?.forEach((added) => {
+      this.eventMap.set(added.projectRoot.fsPath, added);
+    });
+    event.removed?.forEach((removed) => {
+      this.eventMap.delete(removed.projectRoot.fsPath);
+    });
+  }
+
+  async getProblems() {
+    this.telemetry.sendTelemetryEvent("altimateScan");
+    const projects = this.dbtProjectContainer.findAllDBTProjects();
+    const altimateCatalog: AltimateCatalog = await getCatalog(projects);
+    for (const project of projects) {
+      const projectEventMap = this.eventMap.get(project.projectRoot.fsPath);
+      const projectDiagnostics: { [fullFilePath: string]: Diagnostic[] } = {};
+
+      getProjectProblems(
+        altimateCatalog,
+        project,
+        projectEventMap,
+        projectDiagnostics,
+      );
+      for (const [filePath, fileDiagnostics] of Object.entries(
+        projectDiagnostics,
+      )) {
+        project.projectHealth.set(Uri.file(filePath), fileDiagnostics);
+      }
+    }
```

Comment:
This logic should go in a separate class. As we are going to add more checks (probably ran in parallel).